internal/bgp/bgp.go

```diff
@@ -665,3 +679,30 @@ func buildTCPMD5Sig(addr net.IP, key string) tcpmd5sig {
 
 	return t
 }
+
+// localAddressExists returns true if the provided address exists locally on
+// any of the host's network interfaces. The function doesn't return errors and
+// instead returns false if an error has occurred while looking for the
+// address. This function is safe to call if addr is nil or malformed.
+func localAddressExists(addr string) bool {
+	interfaces, err := net.Interfaces()
+	if err != nil {
+		return false
+	}
+
+	for _, i := range interfaces {
+		addresses, err := i.Addrs()
+		if err != nil {
+			return false
+		}
+
+		for _, a := range addresses {
+			ip := strings.Split(a.String(), "/")[0]
+			if net.ParseIP(ip).Equal(net.ParseIP(addr)) {
```

Comment:
You can probably simplify this by using what we are doing for layer2. Just "inspire" from this: https://github.com/metallb/metallb/blob/80bfe1cf45bf4338e6eced11ab7a88d3d2356663/internal/layer2/announcer.go#L85-L89

So, you can just check if it is type IPNet and if it is, just `net.ParseIP()` with the `.IP` field.

The strings container with `/` seems more fragile and inconsistent with other parts of the code :)

Reply:
Nice, thanks. I wasn't satisfied with the `/` thing.